### DISCUSSION TITLE
[FEATURE] Weekly instance resets: Add option to choose day of the week

### DISCUSSION THREAD

**Discussion by steenburgh** [azerothcore/azerothcore-wotlk#3888] [Ideas]
**Is your feature request related to a problem? Please describe.**
Instance resets may be on an inconvenient day of the week for a majority of a given server's players. For example, on my server, we do raids on weekends and instances reset Saturday at 11PM server time, which means we both have to finish by 11pm, and start over from scratch if we play again on Sunday.

**Describe the solution you'd like**
Similar to the `Instance.ResetTimeHour`, I'd like to see an `Instance.WeeklyResetDay` or `Instance.FirstResetDate`, from which all future instance reset times are calculated. Using a `FirstResetDate` may be better as it also covers instances that reset every 3 days like Zul'Gurub.

**Describe alternatives you've considered**
I've dug into the AzerothCore source and found a hacky workaround, which I've documented in [this stack overflow post](https://stackoverflow.com/questions/55312974/how-to-setup-a-day-of-week-for-reset-heroic-instances/64417932#64417932). TL;DR, Instance reset times are calculated based off the very first time the server is started, then maintained through subsequent server restarts.

This leaves us with a few options:
* Have players use the ingame ['Extend Raid Lock' function](https://wowwiki.fandom.com/wiki/Instance_Lock_Extension). Have a GM manually reset instances when needed
* Use the database hackery described on the stackoverflow post << This is what I did
* Permanently set the system time on the server to a few days earlier or later

I've also considered changing `Instance.ResetTimeHour`, but that only lets me move the reset time 4 hours back or 20 hours forward, as the default is `4` (4AM UTC, 11PM in my timezone). I can mitigate the issue, sure, but I cannot solve it.

**Additional context**
I found a setting, `Instance.ResetTimeRelativeTimestamp` which, as documented, seems like it *almost* does what I'm asking for...
```
#
#     Instance.ResetTimeRelativeTimestamp
#        Description: Needed for displaying valid instance reset times in ingame calendar.
#                     This timestamp should be set to a date in the past (midnight) on which
#                     both 3-day and 7-day raids were reset.
#        Default:     1135814400 - (Thu, 29 Dec 2005 00:00:00 GMT - meaning that 7-day raid reset falls on Thursdays,
#                                   while 3-day reset falls on "Thu 29 Dec 2005", "Sun 01 Jan 2006", "Wed 04 Jan 2006", and so on)
```
Unfortunately, I've set it to various numbers, and it does not appear to affect instance reset time, *or the ingame calendar, like it claims to*. The `ResetTimeRelativeTimestamp` setting is indeed used in a function that appears to be sending calendar data back to the user, but it's not used anywhere else in the code. Perhaps we could use this setting to determine reset times, but I don't really know what it's real purpose is.

**Comment by xBarselonx** [comment]
I have the same problem with reset. Maybe there is a solution?

**Comment by Helias** [comment]
based on this https://github.com/azerothcore/mod-instance-reset you could create a similar module that add the conf or implement the conf inside AC
or probably you can just use that module as partial solution

**Reply by xBarselonx** [reply]
> based on this https://github.com/azerothcore/mod-instance-reset you could create a similar module that add the conf or implement the conf inside AC
> or probably you can just use that module as partial solution

Can you add an automatic reset to the server, like on official servers, on Wednesday?

**Reply by Helias** [reply]
I have no time to do it, I am sorry

**Reply by xBarselonx** [reply]
this mod can also solve the issue with resetting inst (https://github.com/azerothcore/mod-npc-services)?
